src/main/java/org/jenkinsci/plugins/authorizeproject/strategy/SpecificUsersAuthorizationStrategy.java

```diff
@@ -260,7 +262,11 @@ protected SpecificUsersAuthorizationStrategy newInstanceWithoutAuthentication(
             if (StringUtils.isBlank(userid)) {
                 throw new FormException("userid must be specified", "userid");
             }
-            
+            // TODO use Jenkins.getInstance().getSecurityRealm().getUserIdStrategy().equals(userid, ACL.SYSTEM.getPrincipal().toString())) once Jenkins 1.566+
+            if (userid.equals(ACL.SYSTEM.getPrincipal())) {
+                throw new FormException(Messages.SpecificUsersAuthorizationStrategy_userid_notSystem(), "userid");
+            }
+

```

Comment:
So the "password" for SYSTEM is hardcoded. Ordinarily that should not be usable as a password, but I suspect there may be some code paths where it could be otherwise used to authenticate as the uber powerful user. The likely defense in such cases is to randomise the password on every query so that you can never have a password match... In such cases the specific user option can never work... Also you want different rules about when the special user can be specified hence needing its own impl